### DISCUSSION TITLE
Upload file to JFrog Artifactory

### DISCUSSION THREAD

**Discussion by shailesh1984** [cake-build/cake#4332] [Q&A] [answered]
How to upload a zip file to JFrog Artifactory Using Cake in Task?

**Comment by JuergenRB** [comment, marked as answer]
To upload a zip file to JFrog Artifactory using Cake, you can follow these steps:

* Install JFrog CLI: Ensure you have the JFrog CLI installed on your build agent / machine. You can download it from the JFrog website or Chocolatey.
* Create a Cake Script: Write a Cake script to handle the upload. Below is an example script

```CSharp
Task("Upload-Zip-To-Artifactory")
    .Does(() =>
    {
        var sourcePath = "./path/to/your/file.zip";
        var targetPath = "your-repo/your-path/file.zip";
        var artifactoryUrl = "https://your-artifactory-url/artifactory";
        var user = "your-username";
        var apiKey = "your-api-key";

        var arguments = $"rt u {sourcePath} {targetPath} --url={artifactoryUrl} --user={user} --access-token={apiKey}";
        StartProcess("jfrog", new ProcessSettings { Arguments = arguments });
    });

RunTarget("Upload-Zip-To-Artifactory");
```
[jFrog Docu](https://docs.jfrog-applications.jfrog.io/jfrog-applications/jfrog-cli/cli-for-jfrog-artifactory/generic-files#uploading-files)
[Cake StartProcess](https://cakebuild.net/api/Cake.Common/ProcessAliases/81E648CC)

**Reply by shailesh1984** [reply]
@JuergenRB Thank you for your quick response.
